src/components/OcButton.spec.js

```diff
@@ -0,0 +1,53 @@
+import OcButton from "./OcButton.vue";
+import {createLocalVue, shallowMount} from "@vue/test-utils";
+const localVue = createLocalVue()
+
+
+describe("oc button", () => {
+  const testSlots = { default: '<p class="text">Test button</p>' }
+  function getWrapperWithProps(props) {
+    return shallowMount(OcButton, { localVue: localVue, propsData: props })
+  }
+  function getWrapperWithTestSlot() {
+    return shallowMount(OcButton, { localVue: localVue, slots: testSlots })
+  }
+
+  describe("emit click event", () => {
+    it("emits click event when click is triggered", () => {
+      const wrapper = getWrapperWithProps({})
+      wrapper.trigger("click")
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper.emitted('click')).toBeTruthy()
+      })
+    })
+    it("does not emits click event when disabled", () => {
+      const wrapper = getWrapperWithProps({
+        disabled: true
+      })
+      wrapper.trigger("click")
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper.emitted('click')).toBeFalsy()
+      })
+    })
+    it("does not emits click event when input is triggered", () => {
```

Comment:
This test doesn't make sense to me. Can you elaborate why this would be a good idea?

Reply:
if a slot has an input, will the wrapper be confused? I added this just to verify the emit for invalid events. I can remove it if this makes no sense